From: Ben Hutchings <ben@decadent.org.uk>:
Subject: tracing: Avoid ABI change in 3.2.42
Date: Sun, 31 Mar 2013 23:48:07 +0100
Forwarded: not-needed

Two new fields were added to struct tracer by commit 613f04a0f51e
'tracing: Prevent buffer overwrite disabled for latency tracers'.

This structure definition is not available to OOT modules and the
functions it's used with are not exported at all, but unfortunately it
ends up affecting the symbol versions of various functions that may be
legitimately used to OOT modules.

Hide the changes from genksyms.
---
--- a/kernel/trace/trace.h
+++ b/kernel/trace/trace.h
@@ -272,13 +272,17 @@ struct tracer {
 	/* If you handled the flag setting, return 0 */
 	int			(*set_flag)(u32 old_flags, u32 bit, int set);
 	/* Return 0 if OK with change, else return non-zero */
+#ifndef __GENKSYMS__
 	int			(*flag_changed)(struct tracer *tracer,
 						u32 mask, int set);
+#endif
 	struct tracer		*next;
 	struct tracer_flags	*flags;
 	int			print_max;
 	int			use_max_tr;
+#ifndef __GENKSYMS__
 	bool			enabled;
+#endif
 };
 
 
